src/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -1264,7 +801,49 @@ internal int IsPublicId(string str)
         private static bool InRange(int value, int start, int end)
         {
             Debug.Assert(start <= end);
-            return unchecked((uint)(value - start) <= (uint)(end - start));
+            return (uint)(value - start) <= (uint)(end - start);
+        }
+
+#if XMLCHARTYPE_GEN_RESOURCE
+        //
+        // Code for generating XmlCharType.bin table and s_PublicIdBitmap
+        //
+        // build command line:  csc XmlCharType.cs /d:XMLCHARTYPE_GEN_RESOURCE
+        //
+        public static void Main( string[] args ) {
```

Comment:
This is the code used to generate the .bin resource file. I think it's worth keeping here.

---
In reply to: [122582177](https://github.com/dotnet/corefx/pull/21155#discussion_r122582177) [](ancestors = 122582177)